vhdbuilder/packer/install-dependencies.sh

```diff
@@ -448,11 +459,22 @@ PATCHED_HYPERKUBE_IMAGES="
 1.17.5.1
 1.18.1.1
 1.18.2.1
+1.19.0-beta.1
 "
 for KUBERNETES_VERSION in ${PATCHED_HYPERKUBE_IMAGES}; do
-  CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/hyperkube:v${KUBERNETES_VERSION}"
-  pullContainerImage "docker" ${CONTAINER_IMAGE}
-  echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
+    ## use hyperkube until 1.19
+    if (($(echo ${KUBERNETES_VERSION} | cut -d"." -f2) < 19)); then
+        CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/hyperkube:v${KUBERNETES_VERSION}"
+        pullContainerImage "docker" ${CONTAINER_IMAGE}
+        echo "  - ${CONTAINER_IMAGE}" >>${VHD_LOGS_FILEPATH}
+    fi
+
+    # from 1.17 onwards start using kube-proxy as well
+    if (($(echo ${KUBERNETES_VERSION} | cut -d"." -f2) >= 17)); then
+        CONTAINER_IMAGE="mcr.microsoft.com/oss/kubernetes/kube-proxy:v${KUBERNETES_VERSION}"
```

Comment:
docker pull mcr.microsoft.com/oss/kubernetes/kube-proxy:v1.18.2.1
Error response from daemon: manifest for mcr.microsoft.com/oss/kubernetes/kube-proxy:v1.18.2.1 not found: manifest unknown: manifest tagged by "v1.18.2.1" is not found

Reply:
I think for kube-proxy we should have separate method , don't mix it with the existing hyperkube 

https://msazure.visualstudio.com/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/53245/Hotfix-Releases